packages/lit-dev-content/samples/examples/react-events/app.tsx

```diff
@@ -1,24 +1,16 @@
-import {React, ReactDOM} from './react.js';
-import {createComponent} from '@lit-labs/react';
-import {CounterButton as CounterButtonWC} from './counter-button.js';
-
-const CounterButton = createComponent({
-  react: React,
-  tagName: 'counter-button',
-  elementClass: CounterButtonWC,
-});
-
-const {useCallback, useState} = React;
+import React, {useCallback, useState} from 'react';
+import {createRoot} from 'react-dom/client';
+import './counter-button.js';
 
 export const App = () => {
   const [count, setCount] = useState(0);
 
-  const onClick = useCallback(() => setCount(count + 1), [count]);
+  const handleClick = useCallback(() => setCount((c) => c + 1), []);
```

Comment:
Spoke offline - here are notes. Although this is a great change, and in line with [React guidance](https://react.dev/reference/react/useCallback#skipping-re-rendering-of-components), in this sample it's not required.

Both `count` and `setCount` are passed to `counter-button`, so memoizing one doesn't provide any advantages - and may provide some overhead.

Per https://react.dev/reference/react/useCallback#skipping-re-rendering-of-components, `useCallback` is not required here.